qa/tasks/mgr/dashboard/test_cephfs.py

```diff
@@ -290,3 +290,16 @@ def test_listing_of_ui_api_ls_on_deeper_levels(self):
         ui_api_ls = self.ui_ls_dir('/pictures', 0)
         self.assertEqual(api_ls, ui_api_ls)
         self.rm_dir('/pictures')
+
+    def test_cephfs_clients_get_after_mds_down(self):
+        fs_id = self.get_fs_id()
+        self._get("/api/cephfs/{}/clients".format(fs_id))
+        self.assertStatus(200)
+
+        self.fs.fail()
+        self._get("/api/cephfs/{}/clients".format(fs_id))
+        self.assertStatus(500)
```

Comment:
I'm a little disappointed it's not `503`. The dashboard could even set `Retry-after` header to indicate a retry timeout. Oh well.

Reply:
@YiteGu are you sure it returns 500 and not 503? I can see places where 503 is returned in dashboard tests.

Reply:
```
2023-07-05T20:39:50.478+0800 7f5700c4e700 10 client.0 fetch_fsmap finished waiting for FSMap version 243
2023-07-05T20:39:50.478+0800 7f5700c4e700 -1 client.0 resolve_mds: targets rank: 0 is down
2023-07-05T20:39:50.487+0800 7f5700c4e700  0 [dashboard ERROR viewcache] Error while calling fn=<function CephFSClients.get at 0x7f5756ed5510> ex=failed to send command to mds: (11) Resource temporarily unavailable
Traceback (most recent call last):
  File "/home/ceph/src/pybind/mgr/dashboard/tools.py", line 147, in run
    val = self.fn(*self.args, **self.kwargs)
  File "/home/ceph/src/pybind/mgr/dashboard/controllers/cephfs.py", line 529, in get
    return CephService.send_command('mds', 'session ls', srv_spec='{0}:0'.format(self.fscid))
  File "/home/ceph/src/pybind/mgr/dashboard/services/ceph_service.py", line 326, in send_command
    mgr.send_command(result, srv_type, srv_spec, json.dumps(argdict), "")
  File "/home/ceph/src/pybind/mgr/mgr_module.py", line 1701, in send_command
    self._ceph_send_command(result, svc_type, svc_id, command, tag, inbuf)
RuntimeError: failed to send command to mds: (11) Resource temporarily unavailable
2023-07-05T20:39:50.487+0800 7f5700c4e700  0 [dashboard DEBUG viewcache] execution of <function CephFSClients.get at 0x7f5756ed5510> finished in: -1688560790.4767783
2023-07-05T20:39:50.491+0800 7f570d627700  0 [volumes INFO mgr_util] scanning for idle connections..
2023-07-05T20:39:50.491+0800 7f570d627700  0 [volumes DEBUG mgr_util] fs_name (a) connections ([])
2023-07-05T20:39:50.491+0800 7f570d627700  0 [volumes INFO mgr_util] cleaning up connections: []
2023-07-05T20:39:50.494+0800 7f56fbb44700  0 [dashboard ERROR exception] Internal Server Error
Traceback (most recent call last):
  File "/home/ceph/src/pybind/mgr/dashboard/services/exception.py", line 47, in dashboard_exception_handler
    return handler(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/cherrypy/_cpdispatch.py", line 54, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/home/ceph/src/pybind/mgr/dashboard/controllers/_base_controller.py", line 263, in inner
    ret = func(*args, **kwargs)
  File "/home/ceph/src/pybind/mgr/dashboard/controllers/_rest_controller.py", line 191, in wrapper
    return func(*vpath, **params)
  File "/home/ceph/src/pybind/mgr/dashboard/controllers/cephfs.py", line 53, in clients
    return self._clients(fs_id)
  File "/home/ceph/src/pybind/mgr/dashboard/controllers/cephfs.py", line 300, in _clients
    status, clients = cephfs_clients.get()
  File "/home/ceph/src/pybind/mgr/dashboard/tools.py", line 245, in wrapper
    return rvc.run(fn, args, kwargs)
  File "/home/ceph/src/pybind/mgr/dashboard/tools.py", line 227, in run
    raise self.exception
  File "/home/ceph/src/pybind/mgr/dashboard/tools.py", line 147, in run
    val = self.fn(*self.args, **self.kwargs)
  File "/home/ceph/src/pybind/mgr/dashboard/controllers/cephfs.py", line 529, in get
    return CephService.send_command('mds', 'session ls', srv_spec='{0}:0'.format(self.fscid))
  File "/home/ceph/src/pybind/mgr/dashboard/services/ceph_service.py", line 326, in send_command
    mgr.send_command(result, srv_type, srv_spec, json.dumps(argdict), "")
  File "/home/ceph/src/pybind/mgr/mgr_module.py", line 1701, in send_command
    self._ceph_send_command(result, svc_type, svc_id, command, tag, inbuf)
RuntimeError: failed to send command to mds: (11) Resource temporarily unavailable
2023-07-05T20:39:50.494+0800 7f56fbb44700 20 mgr get_config  key: mgr/dashboard/debug
2023-07-05T20:39:50.494+0800 7f56fbb44700 10 mgr get_typed_config  debug not found
2023-07-05T20:39:50.495+0800 7f56fbb44700  0 [dashboard ERROR request] [::ffff:127.0.0.1:39658] [GET] [500] [0.022s] [yitegu] [513.0B] /api/cephfs/1/clients
2023-07-05T20:39:50.495+0800 7f56fbb44700  0 [dashboard ERROR request] [b'{"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "request_id": "2d96cec7-2149-4d55-9517-6ebadd9a4aed"}                                                                                                                                                                                                                                                                                                                        ']
```

Reply:
alright